WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2328,6 +2325,53 @@ public void onMediaUploadCancelClicked(String localMediaId) {
     }
 
     @Override
+    public void onMediaDeleted(String localMediaId) {
+        if (!TextUtils.isEmpty(localMediaId)) {
+            // passing false here as we need to keep the media item in case the user wants to undo
+            cancelMediaUpload(StringUtils.stringToInt(localMediaId), false);
+        }
+    }
+
+    private void cancelMediaUpload(int localMediaId, boolean delete) {
+        MediaModel mediaModel = mMediaStore.getMediaWithLocalId(Integer.valueOf(localMediaId));
+        if (mediaModel != null) {
+            CancelMediaPayload payload = new CancelMediaPayload(mSite, mediaModel, delete);
+            mDispatcher.dispatch(MediaActionBuilder.newCancelMediaUploadAction(payload));
+        }
+    }
+
+    @Override
+    public void onUndoMediaCheck(final String undoedContent) {
```

Comment:
What do you think about simplifying this by making Aztec do the work? It seems out of place to have WPAndroid scan the undone content for media. My thought is that `AztecEditorFragment` could have a `onMediaUndeleted()` event which works the same as `onMediaDeleted()` - it simply passes the ID of the deleted media.